integrationTests/relayers/slowTests/framework/ethereumHandler.go

```diff
@@ -391,90 +391,56 @@ func (handler *EthereumHandler) mintTokens(
 	require.Equal(handler, mintAmount.String(), balance.String())
 }
 
-// CreateBatchOnEthereum will create a batch on Ethereum using the provided tokens parameters list
-func (handler *EthereumHandler) CreateBatchOnEthereum(
+// ApproveForToken will approve the spender to spend the amount of tokens on the behalf of the approver
+func (handler *EthereumHandler) ApproveForToken(
 	ctx context.Context,
-	mvxCalleeScAddress core.AddressHandler,
-	tokensParams ...TestTokenParams,
+	token *TokenData,
+	approver KeysHolder,
+	spender common.Address,
+	amount *big.Int,
 ) {
-	for _, params := range tokensParams {
-		handler.createDepositsOnEthereumForToken(ctx, params, handler.AliceKeys, handler.BobKeys, mvxCalleeScAddress)
-	}
-
-	// wait until batch is settled
-	batchSettleLimit, _ := handler.SafeContract.BatchSettleLimit(nil)
-	for i := uint8(0); i < batchSettleLimit+1; i++ {
-		handler.SimulatedChain.Commit()
-	}
-}
-
-// SendFromEthereumToMultiversX will create the deposit transactions on the Ethereum side
-func (handler *EthereumHandler) SendFromEthereumToMultiversX(
-	ctx context.Context,
-	from KeysHolder,
-	to KeysHolder,
-	mvxTestCallerAddress core.AddressHandler,
-	tokensParams ...TestTokenParams,
-) {
-	for _, params := range tokensParams {
-		handler.createDepositsOnEthereumForToken(ctx, params, from, to, mvxTestCallerAddress)
-	}
+	auth, _ := bind.NewKeyedTransactorWithChainID(approver.EthSK, handler.ChainID)
+	tx, err := token.EthErc20Contract.Approve(auth, spender, amount)
+	require.NoError(handler, err)
+	handler.SimulatedChain.Commit()
+	handler.checkEthTxResult(ctx, tx.Hash())
 }
 
-func (handler *EthereumHandler) createDepositsOnEthereumForToken(
+// SendDepositTransactionFromEthereum will send a deposit transaction from Ethereum to MultiversX
+func (handler *EthereumHandler) SendDepositTransactionFromEthereum(
 	ctx context.Context,
-	params TestTokenParams,
 	from KeysHolder,
 	to KeysHolder,
 	targetSCAddress core.AddressHandler,
+	token *TokenData,
+	operation TokenOperations,
 ) {
-	// TODO: transfer only required amount for deposit to the test key
-
-	// add allowance for the sender
 	auth, _ := bind.NewKeyedTransactorWithChainID(from.EthSK, handler.ChainID)
 
-	token := handler.TokensRegistry.GetTokenData(params.AbstractTokenIdentifier)
-	require.NotNil(handler, token)
-	require.NotNil(handler, token.EthErc20Contract)
-
-	allowanceValue := big.NewInt(0)
-	for _, operation := range params.TestOperations {
-		if operation.ValueToTransferToMvx == nil {
-			continue
-		}
-
-		allowanceValue.Add(allowanceValue, operation.ValueToTransferToMvx)
-	}
-
-	if allowanceValue.Cmp(zeroValueBigInt) > 0 {
-		tx, err := token.EthErc20Contract.Approve(auth, handler.SafeAddress, allowanceValue)
-		require.NoError(handler, err)
-		handler.SimulatedChain.Commit()
-		handler.checkEthTxResult(ctx, tx.Hash())
-	}
-
+	var tx *types.Transaction
```

Comment:
I think we still need this early exit:
```go
if operation.ValueToTransferToMvx == nil {
    return
}
```

it was found on L458, on the old code

Reply:
I thought it would be cleaner to keep the logic for deposits isolated within `SendDepositTransactionFromEthereum`. The validation check is now performed beforehand (L468 from TestSetup.go)